index.js

```diff
@@ -47,9 +47,17 @@ function translate(text, opts) {
         };
         data[token.name] = token.value;
 
-        return url + '?' + querystring.stringify(data);
+        var fullUrl = url + '?' + querystring.stringify(data);
+        if (fullUrl.length > 2083) {
+            delete data.q;
+            return [
+                url + '?' + querystring.stringify(data),
+                {method: 'POST', body: {q: text}}
+            ];
+        }
+        return [fullUrl];
     }).then(function (url) {
-        return got(url).then(function (res) {
+        return got.apply(got, url).then(function (res) {
```

Comment:
We can set got's `options` param instead of tricky `apply`.

Reply:
You mean that we can just use `got(...url)` in place of `got.apply`, right? Ye, that was a suggestion to use in the PR this code originally came from anyway, so if you're thinking of that, I could definitely do that.

Reply:
This has already been done. Convo resolved.